lib/charms/operator_libs_linux/v2/snap.py

```diff
@@ -566,7 +583,7 @@ def ensure(
                 self._install(channel, cohort, revision)
             else:
                 # The snap is installed, but we are changing it (e.g., switching channels).
-                self._refresh(channel, cohort, revision)
+                self._refresh(channel, cohort, revision, devmode)
```

Comment:
I realize this isn't something you changed here (you just added to it), but it seems to me asking for trouble not to use keyword arguments here (accidental argument-order problems). Let's do:

```suggestion
                self._refresh(channel=channel, cohort=cohort, revision=revision, devmode=devmode)
```

Reply:
Done